core/controllers/base.py

```diff
@@ -892,14 +899,40 @@ class CsrfTokenManager:
     def init_csrf_secret(cls) -> None:
         """Verify that non-default CSRF secret exists; creates one if not."""
 
+        csrf_value = platform_feature_services.get_platform_parameter_value(
+            CSRF_SECRET.name
+        )
         # Any non-default value is fine.
-        if CSRF_SECRET.value and CSRF_SECRET.value != DEFAULT_CSRF_SECRET:
+        if csrf_value != DEFAULT_CSRF_SECRET:
```

Comment:
Btw I thought of a way to check this is working properly.

- Set a csrf secret, then go to any page.
- In a separate tab, change the csrf secret.
- Try to perform an action on that page that sends a POST/PUT request. The action should be denied because of a CSRF token mismatch.
- After reloading the page, the action should work.

Could you please check that? Thanks!

Reply:
I have updated the proof section in the PR description.